eclair-core/src/main/scala/fr/acinq/eclair/io/Switchboard.scala

```diff
@@ -207,14 +207,14 @@ object Switchboard extends Logging {
     // We are interested in incoming HTLCs, that have been *cross-signed* (otherwise they wouldn't have been relayed).
     // If the HTLC is not in their commitment, it means that we have already fulfilled/failed it and that we can remove
     // the command from the pending relay db.
-    val channel2Htlc: Set[(ByteVector32, Long)] =
-    channels
-      .flatMap(_.commitments.remoteCommit.spec.htlcs)
-      .filter(_.direction == OUT)
-      .map(htlc => (htlc.add.channelId, htlc.add.id))
-      .toSet
-
-    val pendingRelay: Set[(ByteVector32, Long)] = relayDb.listPendingRelay()
+    val channel2Htlc: Seq[(ByteVector32, Long)] = for {
```

Comment:
I don't get why you're changing that method...

Reply:
it was just easier for me to reason what's going on by replacing with method calls with `for`

Reply:
Or, if you mean `Set -> Seq`, it's because `.toSet` in old method was superfluous because it's only used in `val toClean = pendingRelay -- channel2Htlc` where `pendingRelay` is a `Set` already and `channel2Htlc` can be just a `Seq`.